src/app/requests/page.tsx

```diff
@@ -0,0 +1,89 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+"use client";
+
+import PageContainer from "@/components/PageContainer";
+import { faDatabase, faFileText } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  redirect,
+  usePathname,
+  useRouter,
+  useSearchParams,
+} from "next/navigation";
+import ApplicationsPage from "./applications";
+import GrantedDatasetsPage from "./grantedDatasets";
+
+enum ActiveTab {
+  APPLICATIONS = "APPLICATIONS",
+  DATASETS = "GRANTED DATASETS",
+}
+
+function RequestPage() {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const path = usePathname();
+
+  const activeTab: ActiveTab = searchParams.get("tab") as ActiveTab;
+
+  if (
+    !activeTab ||
+    !Object.values(ActiveTab).includes(activeTab.toUpperCase() as ActiveTab)
+  ) {
+    const newPath = `${path}?tab=${ActiveTab.APPLICATIONS.toLowerCase()}`;
+    redirect(newPath);
+  }
+
+  function setActiveTab(e: React.MouseEvent<HTMLButtonElement>) {
+    const newTab = e.currentTarget.textContent!.toLowerCase();
+    if (newTab === activeTab) return;
+
+    const newParams = new URLSearchParams();
+    newParams.set("tab", newTab);
+    router.push(`${path}?${newParams.toString()}`);
+  }
+
+  return (
+    <PageContainer className="pt-6">
+      <div className="flex">
+        <button
```

Comment:
this button can be a local component TabButton defined in same file above and reused